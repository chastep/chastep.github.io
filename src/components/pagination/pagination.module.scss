@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.pagination {
  display: flex;
  @include mixins.margin-top(2);

  .previous {
    text-align: left;
    width: 50%;

    .previousLink {
      color: hsl(var(--color-foreground));
      font-size: 26px;
      font-weight: bold;

      &:hover,
      &:focus {
        color: hsl(var(--color-primary));
      }

      &.disable {
        color: hsl(var(--color-muted-foreground));
        pointer-events: none;
      }
    }
  }

  .next {
    text-align: right;
    width: 50%;

    .nextLink {
      color: hsl(var(--color-foreground));
      font-size: 26px;
      font-weight: bold;

      &:hover,
      &:focus {
        color: hsl(var(--color-primary));
      }

      &.disable {
        color: hsl(var(--color-muted-foreground));
        pointer-events: none;
      }
    }
  }
}
