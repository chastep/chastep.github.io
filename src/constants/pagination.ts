const pagination = {
  prevPage: "← PREV",
  nextPage: "→ NEXT",
};

export { pagination }
