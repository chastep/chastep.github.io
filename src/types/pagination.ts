interface Pagination {
  currentPage: number;
  prevPagePath: string;
  nextPagePath: string;
  hasPrevPage: boolean;
  hasNextPage: boolean;
}

export { type Pagination };
