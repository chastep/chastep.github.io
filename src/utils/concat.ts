const concat = (...args: string[]): string => args.join("");

export { concat };
