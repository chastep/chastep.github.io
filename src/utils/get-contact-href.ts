const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    mastodon: `${contact}`,
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    x: `https://www.x.com/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    bluesky: `https://bsky.app/profile/${contact}.bsky.social`,
  };

  return hrefs[name] ?? contact;
};

export { getContactHref };
