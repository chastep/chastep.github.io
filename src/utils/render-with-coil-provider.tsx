import React, { type ReactElement, type ReactNode, type FC } from "react";
import { CoilProvider } from "@alxshelepenok/diesel";
import { render, type RenderOptions } from "@testing-library/react";
import renderer, { type ReactTestRenderer, type TestRendererOptions } from "react-test-renderer";

interface WithCoilProviderProps {
  children: ReactNode;
}

const WithCoilProvider: FC<WithCoilProviderProps> = ({ children }) => (
  <CoilProvider>{children}</CoilProvider>
);

const renderWithCoilProvider = (
  ui: ReactElement,
  options?: RenderOptions,
) => render(ui, { wrapper: WithCoilProvider, ...options });

export const createSnapshotsRenderer = (
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer =>
  renderer.create(<CoilProvider>{nextElement}</CoilProvider>, options);

export { renderWithCoilProvider };
