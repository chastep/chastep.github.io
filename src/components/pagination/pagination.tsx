import React from "react";

import { Link } from "gatsby";
import classNames from "classnames";

import { pagination } from "@/constants/pagination";

import * as styles from "./pagination.module.scss";

type Props = {
  prevPagePath: string;
  nextPagePath: string;
  hasNextPage: boolean;
  hasPrevPage: boolean;
};

const Pagination = ({
  prevPagePath,
  nextPagePath,
  hasNextPage,
  hasPrevPage,
}: Props) => {
  const prevClassName = classNames(styles.previousLink, {
    [styles.disable]: !hasPrevPage,
  });

  const nextClassName = classNames(styles.nextLink, {
    [styles.disable]: !hasNextPage,
  });

  return (
    <div className={styles.pagination}>
      <div className={styles.previous}>
        <Link
          rel="prev"
          to={hasPrevPage ? prevPagePath : "/"}
          className={prevClassName}
        >
          {pagination.prevPage}
        </Link>
      </div>
      <div className={styles.next}>
        <Link
          rel="next"
          to={hasNextPage ? nextPagePath : "/"}
          className={nextClassName}
        >
          {pagination.nextPage}
        </Link>
      </div>
    </div>
  );
};

export { Pagination };
