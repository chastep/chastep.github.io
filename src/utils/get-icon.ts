import { icons } from "@/constants/icons";

const getIcon = (name: keyof typeof icons) => icons[name] || {};

export { getIcon };
