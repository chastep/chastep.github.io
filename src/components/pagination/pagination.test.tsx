import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { Pagination } from "@/components/pagination/pagination";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("Pagination", () => {
  test("renders correctly", () => {
    const props = { ...mocks.pageContext.pagination };
    const tree = createSnapshotsRenderer(<Pagination {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
