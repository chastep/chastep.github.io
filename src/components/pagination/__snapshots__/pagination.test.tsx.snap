// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "← PREV",
          ],
          "props": {
            "aria-current": undefined,
            "className": "",
            "href": "/typography/page/1",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "rel": "prev",
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "→ NEXT",
          ],
          "props": {
            "aria-current": undefined,
            "className": "",
            "href": "/typography/page/3",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "rel": "next",
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
